source/common/router/weighted_cluster_specifier.cc

```diff
@@ -0,0 +1,321 @@
+#include "source/common/router/weighted_cluster_specifier.h"
+
+#include "source/common/config/well_known_names.h"
+#include "source/common/router/config_utility.h"
+
+namespace Envoy {
+namespace Router {
+
+absl::Status validateWeightedClusterSpecifier(const ClusterWeightProto& cluster) {
+  if (cluster.name().empty() != cluster.cluster_header().empty()) {
+    return absl::OkStatus();
+  }
```

Comment:
It took me a minute to understand what's happening here :D

Should we we drop a comment here with the intent?